python/cudf/cudf/core/dataframe.py

```diff
@@ -1197,7 +1196,13 @@ def reset_index(self, drop=False):
                 out[c] = self[c]
         else:
             out = self
-        return out.set_index(RangeIndex(len(self)))
+        if inplace is True:
+            for column_name in set(out.columns) - set(self.columns):
```

Comment:
`set`s are not ordered - does this mean we might lose existing ordering of the columns?

Reply:
In this case I don't think so. Because `reset_index(drop=False, inplace=True)` creates one new column (with either `index` or the index name as its name), only that column should be moved to the beginning. The existing columns in `self` aren't touched.